addons/controllers/addon_controller.go

```diff
@@ -311,6 +314,9 @@ func (r *AddonReconciler) reconcileNormal(
 		return ctrl.Result{}, kerrors.NewAggregate(errors)
 	}
 
+	if !controllerutil.ContainsFinalizer(cluster, addontypes.AddonFinalizer) {
```

Comment:
Typically I have seen finalizer being added in the beginning of the reconcile cycle so that even if there are any errors in any of the reconcile flows the addon secret would be deleted if cluster is deleted.

Reply:
moved to begining of noramalReconcile